scripts/localizationLambda.js

```diff
@@ -0,0 +1,134 @@
+'use strict';
+
+const validCountryCodes = ['au', 'ca', 'nz', 'uk'];
+const validCountryCodesLength = validCountryCodes.length; // cache this
+
+// dontProcess turns off all processing (ultimately, localization)
+const dontProcess = [
+  /^\/blog\/?/, // don't localize the blog
+  /\.[A-z0-9]{2,4}\/?$/, // don't process files; eg.: xml, html, txt, mp4, etc
+  /\.htaccess\/?$/,
+];
+const dontProcessLength = dontProcess.length;
+
+function isValidCountryCode (code) {
+  if (!code || typeof code !== 'string') return false;
+  code = code.toLowerCase();
+  return (validCountryCodes.indexOf(code) > -1);
+}
+
+function getPathCountryCode (uri) {
+  if (!uri || typeof uri !== 'string' || uri.length < 3) return;
+  uri = uri.toLowerCase();
+
+  for (let i = 0; i < validCountryCodesLength; i++) {
+    const code = validCountryCodes[i];
+    if (
+      code &&
+      uri.indexOf(`/${code}`) === 0 && // eg. startsWith
+      (uri.length === 3 || // if only 3 characters, then it matches 100%
+        (uri.length > 3 && uri.indexOf('/', 1) === 3) // if 4+ length, it must have a trailing slash, eg. /nz/, not /nzd-something
+      )
+    ) {
+      return code; // return found country
+    }
+  }
+}
+
+function getHeaderCountryCode (request) {
+  let code = request.headers['cloudfront-viewer-country'][0].value;
```

Comment:
Is this structure guaranteed? I would add some safe guards just to be sure.

Reply:
I probably should guard everything at the top level as a failure would result in a horrid Cloudfront error page.  This is straight AWS code.

I'll just add a `isRequestValid`, `isResponseValid` type helper.  The problem is, if AWS changes this, we'll never know until we notice it's not redirecting.  I'd hope they never change this.

Reply:
Instead, I just wrapped this in a `try/catch` function... :|